components/Form/AddressLookup/AddressLookup.spec.tsx

```diff
@@ -1,35 +1,294 @@
-import { defaultValidation } from './AddressLookup';
+import { defaultValidation, AddressBox } from './AddressLookup';
+import AddressLookupWrapper from './AddressLookupWrapper';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { addressAPIWrapperFactory } from 'factories/postcode';
+import axios from 'axios';
+
+jest.mock('axios');
+const mockedAxios = axios as jest.Mocked<typeof axios>;
 
 describe('AddressLookup', () => {
   describe('defaultValidation', () => {
-    it('should work properly - if not required', () => {
-      const validate = defaultValidation();
-      expect(validate.address()).toBe(true);
-      expect(validate.address({ address: '' })).toBe(true);
-      expect(validate.address({ address: 'foo' })).toBe(true);
-      expect(validate.postcode()).toBe(true);
-      expect(validate.postcode({ postcode: '' })).toBe(true);
-      expect(validate.postcode({ postcode: 'foo' })).toBe(
-        'You must enter a valid postcode'
-      );
-      expect(validate.postcode({ postcode: 'e83as' })).toBe(true);
-    });
-
-    it('should work properly - if required', () => {
-      const validate = defaultValidation({ required: true });
-      expect(validate.address()).toBe('You must enter an address');
-      expect(validate.address({ address: '' })).toBe(
-        'You must enter an address'
-      );
-      expect(validate.address({ address: 'foo' })).toBe(true);
-      expect(validate.postcode()).toBe('You must enter a valid postcode');
-      expect(validate.postcode({ postcode: '' })).toBe(
-        'You must enter a valid postcode'
-      );
-      expect(validate.postcode({ postcode: 'foo' })).toBe(
-        'You must enter a valid postcode'
-      );
-      expect(validate.postcode({ postcode: 'e83as' })).toBe(true);
+    describe('for address', () => {
+      let validate: {
+        address: (
+          arg0?: Partial<AddressBox['value']>
+        ) => true | 'You must enter an address';
+        postcode: (
+          arg0?: Partial<AddressBox['value']>
+        ) => true | 'You must enter a valid postcode';
+      };
+
+      beforeEach(() => {
+        validate = defaultValidation();
+        return validate;
+      });
+
+      it('when validation is not required should return true when address has no value', () => {
```

Comment:
Might be a personal preference thing but I'd prefer to organise these tests like so describe blocks are:

- "for address"
  - "when validation is not required"
  - "when validation is required"

and then remove the need to have "when validation is not required/ required" in each of the test description as it makes it long to read in the terminal.

Reply:
Thanks, that makes sense, I've rearranged it as you've suggested